frontend/src/app/pages/playlist/playlist.component.ts

```diff
@@ -0,0 +1,92 @@
+import { HttpClient } from '@angular/common/http';
+import { Component } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { MessageService } from 'primeng/api';
+import { Observable, Subscription, finalize, of, switchMap, tap } from 'rxjs';
+import { PlaylistVideosDto } from 'src/app/core/models/playlist-videos-dto';
+import { PlaylistService } from 'src/app/core/services/playlist.service';
+
+@Component({
+  selector: 'app-playlist',
+  templateUrl: './playlist.component.html',
+  styleUrls: ['./playlist.component.scss']
+})
+export class PlaylistComponent {
+  id!: string;
+  playlist!: PlaylistVideosDto;
+  showDialog = false;
+  inputText = '';
+  inputSwitch = false;
+  isProgressSpinnerVisible = false;
+  subscriptions: Subscription[] = [];
+
+  constructor(
+    private route: ActivatedRoute,
+    private httpClient: HttpClient,
+    private playlistService: PlaylistService,
+    private messageService: MessageService,
+    private router: Router) {
+    this.id = this.route.snapshot.paramMap.get('id')!;
+
+    this.getPlaylist();
+  }
+
+  public getPlaylist() {
+    this.playlistService.getPlaylistVideos(this.id).subscribe(playlist => {
+      this.playlist = playlist;
+    });
+  }
+
+  public goToVideo(id: string): void {
+    this.router.navigate(['videos/' + id]);
+  }
+
+  public addVideo() {
```

Comment:
in my opinion we shouldnt have add video button here. We should add the videos to playlist from video page and only from there